suite/tests/linux/clone.c

```diff
@@ -113,10 +108,10 @@ test_thread(bool share_sighand, bool clone_vm, bool use_clone3)
         /* If SYS_clone3 is not defined, we simply use SYS_clone instead, so that
          * the expected output is the same in both cases.
          */
-        child = create_thread(run, NULL, &stack, share_sighand, clone_vm);
+        child = create_thread(run_with_exit, &stack, share_sighand, clone_vm);
```

Comment:
The raw clone syscall takes no function; the glibc wrapper takes a function and appends an exit syscall to it.  Looks like dynamorio_clone is in between, taking a function but requiring the user to have it end with exit?  Should we update the comments for dynamorio_clone and create_thread here to make this clear?

Reply:
Add a comment to dynamorio_clone. I think create_thread can live without one.